src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -216,11 +239,25 @@ public void deleteWorkspace(String authToken, UUID id) {
   }
 
   @Traced
-  public boolean isAuthorized(
+  public boolean isAuthorizedWrapped(
```

Comment:
nit: can we develop some more precise terminology? `Wrapped` isn't very informative.
Alternatively, consider a helper function on SamService to catch and rethrow interrupted exceptions instead of doubling the API space. Little more verbose at call sites, but more explicit. Or just suffer the try-catch everywhere.
Or just use the static as a private helper in SamService.
```
public static <T> rethrowIfSamInterrupted(InterruptedSupplier<T> function, String operation) {
  try {
    return function.apply();
  } catch (InterruptedException e) {
    throw SamExceptionFactory.create("Interrupted during Sam operation " + operation, e);
  }
}

// at call site instead of
samService.isAuthorizedWrapped(accessToken, iamResourceType, resourceId, action);
// wrap
rethrowIfSamInterrupted(
  () -> samService.isAuthorized(accessToken, iamResourceType, resourceId, action), "isAuthorized");
```
I'm not convinced this is a good idea, but the double API method seems easy to use in the wrong place.

Reply:
I like this approach, adopted.